test/functorch/test_eager_transforms.py

```diff
@@ -3456,10 +3463,9 @@ def compute_loss(weights, batch, targets):
                 return loss
 
             grad_weights, loss = grad_and_value(compute_loss)(weights, batch, targets)
-            new_weights = []
-            with torch.no_grad():
-                for grad_weight, weight in zip(grad_weights, weights):
-                    new_weights.append(weight - grad_weight * lr)
+            new_weights = self._update_params(weights, grad_weights, lr, mechanism)
+            if mechanism != "make_functional":
+                new_weights = list(new_weights.values())
             # NB: return looks weird because torch.vmap must return Tensors
             return (loss, *new_weights)
```

Comment:
(no action required) this comment looks outdated, right?

Reply:
Yep changed this and made it work with pytrees